src_c/camera_v4l2.c

```diff
@@ -35,8 +35,15 @@ v4l2_list_cameras(int *num_devices)
     num = *num_devices;
 
     devices = (char **)malloc(sizeof(char *) * 65);
+    if (!devices) {
+        return NULL;
+    }
 
     device = (char *)malloc(sizeof(char) * 13);
+    if (!device) {
+        return NULL;
```

Comment:
Looks like there's a leak here at least with devices needing to be freed.

Reply:
Good catch

Reply:
Ah looks like there are more leaks in this function, gonna fix all of em